code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -143,6 +150,82 @@
 			if(istype(user, /mob/living/carbon/human) && src.loc == user)
 				var/mob/living/carbon/human/H = user
 				H.update_inv_hands()
+	if(istype(A, /obj/item/stack/rods))
+		to_chat(user, "<span class='notice'>You start jamming a rod into \the [src]'s barrels to try and loosen the shells...</span>")
+		if(do_after(user, src, 60))
+			jammed = 0
+			to_chat(user, "<span class='notice'>You manage to loosen the shells.</span>")
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/verb/toggle_doubleshot()
+	set name = "Toggle Shooting Both Barrels"
+	set category = "Object"
+	if((world.time >= last_fired + fire_delay) && !doubleshooting)
+		doubleshot = !doubleshot
+		if(!doubleshot)
+			fire_delay = initial(fire_delay)
+		to_chat(usr, "You switch \the [src]'s fire selector to [doubleshot ? "fire both barrels at once" : "fire one barrel at a time"].")
+
+#define JAMMED 1
+#define DROPGUN 2
+#define KNOCKDOWN 3
+#define RECOILBRUISE 4
+
+/obj/item/weapon/gun/projectile/shotgun/doublebarrel/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	if(doubleshot && (getLiveAmmo() == 2)) //ANGERY
+		var/atom/reverse = locate(2*user.x - target.x, 2*user.y - target.y, target.z)
```

Comment:
If this is only needed for one thing that's behind a prob check, why not have it defined down there?